.pre-commit-config.yaml

```diff
@@ -0,0 +1,17 @@
+repos:
+- repo: https://github.com/ambv/black
+  rev: 18.9b0
+  hooks:
+  - id: black
+    args: [--safe, --quiet]
+    language_version: python3
+- repo: https://gitlab.com/pycqa/flake8
```

Comment:
Wow, black & flake8 formatting? 😮 Do they not contradict each other sometimes?

Reply:
Right? They get along pretty well, Black being really an automatically applied, opinionated superset of rules on top of flake8, but we do set some flake8 rules in the `tox.ini` files linting config to make sure they play along nicely, especially with line lengths:

```
[flake8]
exclude = .git, .tox, __pycache__, old, build, dist, txt, .ini
application-import-names = flake8

select = B,C,E,F,W,T4,B9
ignore =
    E501  # Lines are too long.
    W503  # Line break before binary operator.
    E203  # Whitespace before ':'.

import-order-style = pep8
```